src/go/rpk/pkg/api/admin/admin.go

```diff
@@ -185,39 +215,112 @@ func (a *AdminAPI) GetLeaderID() (*int, error) {
 
 // sendAny sends a single request to one of the client's urls and unmarshals
 // the body into into, which is expected to be a pointer to a struct.
+//
+// On errors, this function will keep trying all the nodes we know about until
+// one of them succeeds, or we run out of nodes.  In the latter case, we will return
+// the error from the last node we tried.
 func (a *AdminAPI) sendAny(method, path string, body, into interface{}) error {
-	pick := rng(len(a.urls))
-	url := a.urls[pick] + path
-	res, err := a.sendAndReceive(context.Background(), method, url, body)
-	if err != nil {
-		return err
+	// Shuffle the list of URLs
+	rng := rand.New(rand.NewSource(time.Now().UnixNano()))
+	shuffled := make([]string, len(a.urls))
+	copy(shuffled, a.urls)
+	rng.Shuffle(len(shuffled), func(i, j int) { shuffled[i], shuffled[j] = shuffled[j], shuffled[i] })
+
+	var err error
+	for i := range shuffled {
+		url := shuffled[i] + path
+		var res *http.Response
+
+		// Where there are multiple nodes, disable the HTTP request retry in favour of our
+		// own retry across the available nodes
+		retryable := len(shuffled) == 1
+
+		res, err = a.sendAndReceive(context.Background(), method, url, body, retryable)
+		if err == nil {
+			// Success, return the result from this node.
+			return maybeUnmarshalRespInto(method, url, res, into)
+		} else {
+			if i != len(shuffled)-1 {
+				// We will retry on another node, log the error from this node.
+				log.Infof("Request error, trying another node: %s", err.Error())
```

Comment:
Another option for this `if` block would be to move it to the top of the loop,
```go
if i > 0 {
  log
```
either is fine, I personally prefer extra-loop-logic at the start so I know conditional logic up front, but YMMV and it's pretty inconsequential here.

Reply:
Good idea - I pushed the log up to the top of the loop and made it conditional on `err != nil`, that way there's no reasoning about loop counters at all.